### DISCUSSION TITLE
Coordinating program counter with listing

### DISCUSSION THREAD

**Discussion by astrelsky** [NationalSecurityAgency/ghidra#2618] [Debugger]
I seem to be having trouble with coordinating the program counter with a location in Ghidra. I would expect the current position in the listing to jump to the next instruction when stepping with the debugger. Even when "Track Program Counter" is set this doesn't seem to occur. I can tell that the debugger is functioning by observing the RIP value and its changes as I step over with the debugger.

It would also be a good idea to have some sort of synchronization between the addresses Ghidra displays in the listing and the actual address determined by the module's base address. It is currently impossible to know where you currently are without manually calculating it.

**Comment by astrelsky** [comment]
It appears this was due to an issue with an older jna version.

**Reply by d-millar** [reply]
Let us know if this persists with a newer JNA - we have a related issue.  Fix should be pushed soon.

**Reply by astrelsky** [reply]
> Let us know if this persists with a newer JNA - we have a related issue. Fix should be pushed soon.

It functions more as expected of a debugger with jna:5.4.0
